Add segment analytics docs. (#1579)
RedHatInsights/frontend-components: packages/docs/pages/analytics/index.mdx

import ExternalIconAlt from '@patternfly/react-icons/dist/js/icons/external-link-alt-icon';
import { Alert } from '@patternfly/react-core'

# Analytics

> The chrome provides a Segment analytics  that modules can use to customize and track user events. Before you start using the API, please read mroe about the <a href="https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/#basic-tracking-methods" target="_blank">Segment tracking <ExternalIconAlt/>.</a>

Reviewer: typo
```suggestion
The chrome provides a Segment analytics  that modules can use to customize and track user events. Before you start using the API, please read more about the <a href="https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/#basic-tracking-methods" target="_blank">Segment tracking <ExternalIconAlt/>.</a>
```

## Customizing default page event

*<a href="https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/#page" target="_blank">The page event <ExternalIconAlt /></a> is automatically called on each browser URL change.*

<Alert className="pf-u-mt-md" variant="info" title="Initial page event data cannot be updated!">
  The initial page event is triggered before the module is fully initialized. Additional events can be triggered at runtime at any time.
</Alert>

> ### Access the page event setter trough chrome

Reviewer 2: ```suggestion
### Access the page event setter through chrome
```

    "url": "https://gov.console.stage.redhat.com/insights/vulnerability/cves?affecting=true&page=1&page_size=20&sort=-cvss_score"
  },
  "receivedAt": "2022-07-07T09:05:08.032Z",
  "sentAt": "2022-07-07T09:05:07.998Z",
  "timestamp": "2022-07-07T09:05:08.032Z",
  "type": "page",
  "userId": "51834776"
```

## Custom event tracking

Chrome API also exposes the full `analytics` instance to allow custom event tracking.

Checkout the <a href="https://github.com/RedHatInsights/insights-rbac-ui/pull/1188" target="_blank">RBAC UI example <ExternalIconAlt />.</a>

### Example: tracking user click event

```jsx
import useChrome from '@redhat-cloud-services/frontend-components/useChrome';

...

const RemoveEntityButton = ({ entityId, handleRemove }) => {
  const { analytics } = useChrome();

  return (
    <div>
      <button onClick={() => {
        analytics.track('remove-entity', {
          entityId
        });
        handleRemove()
      }}>Remove entity: {entityId}</button>
    </div>
  )
}

```

## Creating custom UI module segment

*By default, UI modules share the `Hybrid Cloud Console Dev` and `Hybrid Cloud Console` segments.*

> The segment analytics allows definig multiple sources/namespaces to separate (segment) analytics data. To create a new source for a module, it is necessary to provide a unique API key. It can be existing key or new key.

Reviewer 2: ```suggestion
The segment analytics allows defining multiple sources/namespaces to separate (segment) analytics data. To create a new source for a module, it is necessary to provide a unique API key. It can be an existing key or new key.
```
